src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +27,37 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
-                    dependency.Telemetry.Type = "Azure.Functions";
-                    activity.SetCustomProperty("_depTel", dependency);
+                    dependency.Telemetry.Type = DependencyTypeInProc; // Required for proper rendering in App Insights.
+                    activity.SetCustomProperty(DependencyTelemetryKey, dependency);
                 },
                 ActivityStopped = activity =>
                 {
-                    var dependency = activity.GetCustomProperty("_depTel") as IOperationHolder<DependencyTelemetry>;
-                    _telemetryClient.StopOperation(dependency);
-                    dependency?.Dispose();
+                    // Check for Exceptions events
+                    foreach (ActivityEvent activityEvent in activity.Events)
+                    {
+                        TrackExceptionTelemetryFromActivityEvent(activityEvent, _telemetryClient);
+                    }
+
+                    if (activity.GetCustomProperty(DependencyTelemetryKey) is IOperationHolder<DependencyTelemetry> dependencyHolder)
+                    {
+                        var dependency = dependencyHolder.Telemetry;
+
+                        foreach (var item in activity.Tags)
+                        {
+                            if (!dependency.Properties.ContainsKey(item.Key))
```

Comment:
Are there any special tags we should be looking out for and be set on an explicit telemetry property? Such as `ResultCode`, `Data`, or `Target` (for `DependencyTelemetry`).

Reply:
So that a customer can add a tag with `Target` and it'd overwrite whatever it was set to? I think we can see if that's something people want and come back to it -- or they can likely add their own Initializer for that as well.

Reply:
Ah yeah they can do their own initializer true. That is good enough for me.